app/Module.scala

```diff
@@ -16,51 +16,27 @@
 
 import akka.actor.ActorSystem
 import com.google.inject.{AbstractModule, Provides}
-import com.google.inject.name.Names
-import javax.inject.{Named, Provider}
-import play.api.Mode.Mode
+import javax.inject.Named
 import play.api.{Configuration, Environment}
-import play.modules.reactivemongo.MongoDbConnection
-import uk.gov.hmrc.auth.core.AuthConnector
-import uk.gov.hmrc.play.audit.http.config.AuditingConfig
+import uk.gov.hmrc.auth.core.{AuthConnector, AuthorisedFunctions}
 import uk.gov.hmrc.play.audit.model.Audit
 import uk.gov.hmrc.play.bootstrap.auth.DefaultAuthConnector
-import uk.gov.hmrc.play.bootstrap.config.LoadAuditingConfig
 import uk.gov.hmrc.play.bootstrap.http.{DefaultHttpClient, HttpClient}
-import uk.gov.hmrc.play.config.{AppName, ServicesConfig}
 import uk.gov.hmrc.taxhistory._
+import uk.gov.hmrc.taxhistory.config.{AppConfig, DefaultAppConfig}
 import uk.gov.hmrc.taxhistory.services.{PayeCacheService, TaxHistoryMongoCacheService}
 import uk.gov.hmrc.taxhistory.utils.Retry
 
 import scala.concurrent.duration._
 
-class Module(val environment: Environment, val configuration: Configuration) extends AbstractModule with ServicesConfig {
-
-  override protected def mode: Mode = environment.mode
-  override protected def runModeConfiguration: Configuration = configuration
+class Module (val environment: Environment, val configuration: Configuration) extends AbstractModule {
 
   def configure() = {
-
     bind(classOf[HttpClient]).to(classOf[DefaultHttpClient])
-    bind(classOf[AuditingConfig]).toProvider(new AuditingConfigProvider("auditing"))
     bind(classOf[AuthConnector]).to(classOf[DefaultAuthConnector])
     bind(classOf[Audit]).to(classOf[MicroserviceAudit])
     bind(classOf[PayeCacheService]).to(classOf[TaxHistoryMongoCacheService])
-
-    bindConfigInt("mongodb.cache.expire.seconds", default = Some(60 * 30))
-    bindConfigString("mongodb.name")
-    bindConfigString("microservice.services.des.env",                default = Some("local"))
-    bindConfigString("microservice.services.des.authorizationToken", default = Some("local"))
-    bindConfigBoolean("featureFlags.currentYearFlag")
-    bindConfigBoolean("featureFlags.statePensionFlag")
-    bindConfigBoolean("featureFlags.jobSeekersAllowanceFlag")
-
-    bind(classOf[String]).annotatedWith(Names.named("des-base-url")).toProvider(provide(baseUrl("des")))
-    bind(classOf[String]).annotatedWith(Names.named("citizen-details-base-url")).toProvider(provide(baseUrl("citizen-details")))
-
-    bind(classOf[MongoDbConnection]).toProvider(provide(new MongoDbConnection {}))
-
-    bind(classOf[String]).annotatedWith(Names.named("appName")).toProvider(AppNameProvider)
+    bind(classOf[AppConfig]).to(classOf[DefaultAppConfig]).asEagerSingleton()
```

Comment:
Is this module file needed?  Some of these can be brought in via your application.conf or just been made into a singleton class rather than being a trait class pairing.
```
play.modules.enabled += "uk.gov.hmrc.play.bootstrap.AuditModule"
play.modules.enabled += "uk.gov.hmrc.play.bootstrap.HttpClientModule"
play.modules.enabled += "uk.gov.hmrc.play.bootstrap.AuthModule"
```

Having a quick look, the MicroserviceAudit isn't used so could be deleted?

Reply:
Good call yeah, will move it over